### ISSUE TITLE
Grammar expansions for declarations and definitions

### ISSUE THREAD

**Issue by jleeothon** [cortoproject/corto-language#3]
Declarations and definitions and expressions in the corto grammar seem awkwardly intermingled.

Some rough concepts:
- Statements are orders.
- Expressions are operations that make part of statements.
- Expressions can be nested, to make more complex values (take, `1 + 3 * 4`).
- Statements can be nested, e.g. for flow control.

Current grammar expands in the following way (except):

```
statement : expression | declaration ;

expression : declarationExpression ;

declarationExpression : declaration ':' initializer
                      | declaration '{' initializer '}'
                      | declaration '=' commaExpression
                      ;

declaration : identifier identifier+
//            ^ type     ^ names
```

As a result, here is an example of how  a declaration and a definition would be expanded:

```
A simple declaration is expanded:
"int8 a"
statement
declaration
identifier identifier

A definition is expanded
"int8 a: 9"
statement
expression
declaration : initializer
identifier identifier : initializer
```

That doesn't sound reasonable. Both expansions should be on a `declaration` (and/or `definition`) kind of rule, instead of an expression; what's more, we shouldn't use definitions as values inside another expression, e.g. (`(int8 a: 9) + 7`). Note, however, that `(int8 a: 9) + 7` is actually not recognized by the Corto parser. Parenthesis expressions (`bracket_expr`) do only parenthesise around `wait_expr` and not full expressions (that is, `wait_expr` is one level "below" `declaration_expr`).

In summary:
- Named declarations should be statements (they are already).
- Named definitions should be statements (they're not, they are expressions).
- Declarations + assignments should be statements.
- It's okay for (lone) assignments to be expressions.
- Anonymous definitions (curly brace expressions) should be expressions, both when the type is explicit and implicit (e.g. explicit: `vector{1, 3} * vector{4, 5}`; implicit: `Line a: {1, 2}, {3, 4}`).

Questions:
- Why do declarationsExpressions with "=" have a comma expression in the right instead of an initializer?
